tests/e2e/support/objects/app-files/link/actions.ts

```diff
@@ -175,3 +196,21 @@ export const deleteLink = async (args: deleteLinkArgs): Promise<void> => {
   const message = await page.locator(linkUpdateDialog).textContent()
   expect(message.trim()).toBe('Link was deleted successfully')
 }
+
+export const checkpublicLinkVisibility = async (
+  args: checkpublicLinkVisibilityArgs
+): Promise<void> => {
+  const { page, linkUrl, linkName } = args
+  await sidebar.open({ page: page })
+  await sidebar.openPanel({ page: page, name: 'space-share' })
+  const url = await page.locator(util.format(publicLink, linkName)).textContent()
+  expect(url).toBe(linkUrl)
+}
```

Comment:
```suggestion
export const getPublicLinkVisibility = async (
  args: checkpublicLinkVisibilityArgs
): Promise<void> => {
  const { page, linkUrl, linkName } = args
  await sidebar.open({ page: page })
  await sidebar.openPanel({ page: page, name: 'space-share' })
  await page.locator(util.format(publicLink, linkName)).textContent()
  // some calculations
  return true // or false
}
```
PO should only return the state of the Page we're traveling. Make the assertions inside the context (step definitions)